src/transport_ofi.h

```diff
@@ -1349,17 +1326,23 @@ static inline
 int shmem_transport_atomic_supported(shm_internal_op_t op,
                                      shm_internal_datatype_t datatype)
 {
+#ifdef USE_SHR_ATOMICS
+    /* FIXME: Force shared memory atomics build to use software reductions */
```

Comment:
Just curious what needs to happen to support collectives?  `atomicv` I presume - anything else?

Reply:
Yep, just `atomicv`. Simplest thing would be to make this just call the relevant atomic function in a loop. I don't recall a technical reason why I omitted this; I think I just ran out of time.